File: Server/WarframeProgressTrackerApi/WarframeProgressTrackerApi/ViewModels/CollectibleSearchForm.cs
namespace WarframeProgressTrackerApi.ViewModels {
    public class CollectibleSearchForm {
        public bool IncludeFrames { get; set; }
        public bool IncludePrimaryWeapons { get; set; }
        public bool IncludeSecondaryWeapons { get; set; }
        public bool IncludeMeleeWeapons { get; set; }
        public bool IncludePets { get; set; }
        public bool IncludeRoboWeapons { get; set; }
        public bool IncludeOperatorAmps { get; set; }
        public bool IncludeArchwings { get; set; }
        public bool IncludeArchGuns { get; set; }
        public bool IncludeArchMeleeWeapons { get; set; }

        public bool OnlyOnWishlist { get; set; } = false;
        public bool IncludeOwned { get; set; } = true;
        public bool IncludeMastered { get; set; } = true;

        public string SearchText { get; set; }
    }
}

File: Application/warframe-progress-tracker/src/app/_interfaces/collectibleSearchForm.ts
export interface CollectibleSearchForm {
   includeFrames: boolean,
   includePrimaryWeapons: boolean,
   includeSecondaryWeapons: boolean,
   includeMeleeWeapons: boolean,
   includePets: boolean
   includeRoboWeapons: boolean,
   includeOperatorAmps: boolean,
   includeArchwings: boolean,
   includeArchGuns: boolean,
   includeArchMeleeWeapons: boolean,

   onlyOnWishlist: boolean,
   includeOwned: boolean,
   includeMastered: boolean,

   searchText: string
}